### ISSUE TITLE
CameraDisplay visualisation inaccurate for an offset NominalFrame

### ISSUE THREAD

**Issue by vmharvey** [cta-observatory/ctapipe#2028]
**Describe the bug**
When using CameraDisplay to visualise an image in a NominalFrame that has an origin anywhere but at the exact centre of the camera, pixel positions seem generally correct but their sizes are greatly increased to the point they begin obscuring each other.

**To Reproduce**
Here's a relatively tame example of a telescope pointing at 10° altitude and trying to visualise in a NominalFrame with origin at 0°.
```python
import astropy.units as u
from astropy.coordinates import SkyCoord, AltAz
from ctapipe.coordinates import CameraFrame, TelescopeFrame, NominalFrame
from ctapipe.image.toymodel import Gaussian
from ctapipe.instrument import CameraGeometry
from ctapipe.visualization import CameraDisplay
from matplotlib import pyplot as plt

camGeom = CameraGeometry.from_name('FlashCam')

# Set up example physical parameters on the CameraGeometry
focal = 16*u.m
pointing = SkyCoord(alt = 10*u.deg, az = 0*u.deg, frame = AltAz())
camGeom.frame = CameraFrame(focal_length = focal, telescope_pointing = pointing)

# Image data
r = camGeom.guess_radius()
model = Gaussian(x = 0.25*r, y = 0.75*r, width = 0.05*r, length = 0.1*r, psi = '10d')
_,image,_ = model.generate_image(camGeom, intensity = 5000)

# Camera geometry into alternate coordinate systems
horizon = SkyCoord(alt = 0*u.deg, az = 0*u.deg, frame = AltAz())
camGeom_t = camGeom.transform_to(TelescopeFrame(telescope_pointing = pointing))
camGeom_n = camGeom.transform_to(NominalFrame(origin = horizon))

fig,(ax1,ax2) = plt.subplots(1,2)
fig.set_size_inches(16,6)
camDisp_t = CameraDisplay(camGeom_t, ax = ax1, image = image)
camDisp_n = CameraDisplay(camGeom_n, ax = ax2, image = image)
plt.show()
```
![img](https://user-images.githubusercontent.com/95004367/179925415-e348ff6f-a427-4074-98c6-0b71d8bf52e7.png)

**Expected behaviour**
Obviously I do expect there to be a distortion of the image as the pointing direction of the telescope gets further away from the origin (the horizon in this case) due to the projection to 2D, but I had anticipated that this would be correctly visualised with pixels changing their shapes. Instead, they seem to stay as a fixed shape and their sizes become wildly incorrect.

**Supporting information**
ctapipe v0.15.0

**Additional context**
I assume visualisation of NominalFrames like this wasn't really foreseen, which I understand. But I was trying to visualise the data in this frame to confirm that the NominalFrame transform was actually doing what I expected. For reasons like this I think it will happen that users attempt to visualise such frames. If meaningful calculation of the projected pixel shapes isn't possible, perhaps ensuring they at least don't blow up in size would be enough to make the visualisation useful.

**Comment by maxnoe** [comment]
> Obviously I do expect there to be a distortion of the image as the pointing direction of the telescope gets further away from the origin (the horizon in this case) due to the projection to 2D, but I had anticipated that this would be correctly visualised with pixels changing their shapes. Instead, they seem to stay as a fixed shape and their sizes become wildly incorrect.

The reason is that when transforming to an angle based frame from the distance based frame we scale the pixel area by scaling without checking for translation.

Will try to fix!

**Comment by maxnoe** [comment]
@vmharvey mmh. Indeed.. That should probably not happen

**Comment by kosack** [comment]
Just don't try to make this kind of plot at an offset of +/-90 degrees in elevation ;-)  Then you'll really run into problems

**Comment by maxnoe** [comment]
@kosack I think this is not about the spreading (which is indeed correct) but about the slide rotation of the individual pixels

**Comment by kosack** [comment]
Ah, yes that does seem to be a bug. Was not obvious until I zoomed in.

**Comment by maxnoe** [comment]
> Ah, yes that does seem to be a bug. Was not obvious until I zoomed in.

I think I know where this comes from, let me check...

**Comment by maxnoe** [comment]
I was correct, this should fix it: https://github.com/cta-observatory/ctapipe/pull/2037

Side note: I am not completely sure that when transforming between `TelescopeFrame` and `NominalFrame`, scaling the pixel area is the correct thing to do, since the fov of a pixel should not get larger or smaller.

So I guess we should add a check for if we switch units and only then do the scaling? But that would also not work for transforming from `TelescopeFrame` to `NominalFrame`... 

Also, for these high offset versions, the pixel rotation needs to per pixel I guess...

So I'd probably say that at the moment the CameraDisplay (or also the transformations) are not really intended to work for large rotations.

**Comment by maxnoe** [comment]
Fixed in #2037
